lib/AST/ASTContext.cpp

```diff
@@ -3136,6 +3136,16 @@ getGenericFunctionRecursiveProperties(Type Input, Type Result) {
   return properties;
 }
 
+AnyFunctionType::AnyFunctionType(TypeKind Kind, const ASTContext *CanTypeContext,
+                ArrayRef<CallArgParam> Input, Type RawInput, Type Output,
+                RecursiveTypeProperties properties,
+                const ExtInfo &Info)
+: TypeBase(Kind, CanTypeContext, properties),
+  Input((CanTypeContext ?: &RawInput->getASTContext())->AllocateCopy(Input)),
```

Comment:
Hmm... I'm confused why we're doing the `CanTypeContext` null check dance here. If we do the tail allocation thing, then don't need to do the AllocateCopy dance here... we just copy over the data we get.